deployment/cfn/application.py

```diff
@@ -0,0 +1,363 @@
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [GRAPHITE, POSTGRESQL, REDIS, STATSITE, RELP]
+            ] + [
+                ec2.SecurityGroupRule(
+                    IpProtocol='udp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [STATSITE]
+            ] + [
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=ALLOW_ALL_CIDR, FromPort=p,
+                    ToPort=p
+                )
+                for p in [HTTP]
+            ],
+            Tags=self.get_tags(Name=app_server_security_group_name)
+        ))
+
+        return app_server_lb_security_group, app_server_security_group
+
+    def create_load_balancer(self, app_server_lb_security_group):
+        app_server_lb_name = 'elbAppServer'
+
+        return self.add_resource(elb.LoadBalancer(
+            app_server_lb_name,
+            ConnectionDrainingPolicy=elb.ConnectionDrainingPolicy(
+                Enabled=True,
+                Timeout=300,
+            ),
+            CrossZone=True,
+            SecurityGroups=[Ref(app_server_lb_security_group)],
+            Listeners=[
+                elb.Listener(
+                    LoadBalancerPort='80',
+                    InstancePort='80',
+                    Protocol='HTTP',
+                ),
+                elb.Listener(
+                    LoadBalancerPort='443',
+                    InstancePort='80',
+                    Protocol='HTTPS',
+                    SSLCertificateId=Ref(self.ssl_certificate_arn)
+                )
+            ],
+            HealthCheck=elb.HealthCheck(
+                Target='HTTP:80/health-check/',
+                HealthyThreshold='3',
+                UnhealthyThreshold='2',
+                Interval='30',
+                Timeout='5',
+            ),
+            Subnets=Ref(self.public_subnets),
+            Tags=self.get_tags(Name=app_server_lb_name)
+        ))
+
+    def create_auto_scaling_resources(self, app_server_security_group,
+                                      app_server_lb):
+        app_server_launch_config_name = 'lcAppServer'
+
+        app_server_launch_config = self.add_resource(
+            asg.LaunchConfiguration(
+                app_server_launch_config_name,
+                ImageId=Ref(self.app_server_ami),
+                IamInstanceProfile=Ref(self.app_server_instance_profile),
+                InstanceType=Ref(self.app_server_instance_type),
+                KeyName=Ref(self.keyname),
+                SecurityGroups=[Ref(app_server_security_group)]
+            ))
+
+        app_server_auto_scaling_group_name = 'asgAppServer'
+
+        self.add_resource(
+            asg.AutoScalingGroup(
+                app_server_auto_scaling_group_name,
+                AvailabilityZones=Ref(self.availability_zones),
+                Cooldown=300,
```

Comment:
I think so. There is usually room for tuning based on application type, but this is a safe default (it is also the default Amazon uses if you don't provide a `Cooldown`).